daemon/src/model.rs

```diff
@@ -692,17 +690,145 @@ impl str::FromStr for OpeningFee {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         let amount_sat: u64 = s.parse()?;
-        Ok(OpeningFee { amount_sat })
+        Ok(OpeningFee {
+            fee: Amount::from_sat(amount_sat),
+        })
     }
 }
 
+impl_sqlx_type_display_from_str!(OpeningFee);
+
 impl Default for OpeningFee {
     fn default() -> Self {
-        Self::new(Amount::ZERO)
+        Self { fee: Amount::ZERO }
     }
 }
 
-impl_sqlx_type_display_from_str!(OpeningFee);
+impl From<OpeningFee> for Amount {
+    fn from(opening_fee: OpeningFee) -> Self {
+        opening_fee.fee
+    }
+}
```

Comment:
Why are we exposing this? Not ideal IMO.

Reply:
why?

Reply:
Because it is a public API that allows you to unpack the newtype, thus circumventing any invariants that are attached to it. What is the point of using a newtype in the first place then? The newtype conveys what this value means, if you unpack it, that meaning is removed and you can do whatever you want with it, thus removing the ability of the compiler to assist you in implementing the logic correctly.

This shouldn't exist and we should instead call more high-level APIs for the usecase where this is needed.

Reply:
`OpeningFee` is the internal representation. In `projection::CfdOrder` we pass it out as `Amount` - which is not incorrect I think. The projection "flattens" this, it is a Bitcoin amount after all. 

Instead of implementing `From` should I then just add something like `into_amount()` on `OpeningFee`?

Reply:
We agreed on `to_inner` that consumes `self`